treeherder/log_parser/parsers.py

```diff
@@ -422,6 +426,15 @@ def add(self, line, lineno):
 
     def parse_line(self, line, lineno):
         """Check a single line for an error.  Keeps track of the linenumber"""
+        # First line of TaskCluster logs almost certainly has this.
+        if line.startswith('[taskcluster '):
+            self.is_taskcluster = True
+
+        # For performance reasons, only do this if we have identified as
+        # a TC task.
```

Comment:
Did you measure the performance with and without this? From experience, I doubt this is much of a bottleneck.